src/storage-client/src/controller.rs

```diff
@@ -1321,6 +1330,12 @@ where
             if !update.is_empty() {
                 read_capability_changes.insert(id, update);
             }
+            if let Some(new_upper) = new_upper {
+                if new_upper.is_empty() {
+                    self.append(vec![(id, Vec::new(), new_upper.clone())])
```

Comment:
not sure if `update_write_frontiers` is where the logic to actually append the `[]` update ought to live

Reply:
`update_write_frontiers` is passive, it reacts to frontier updates that writers of collections report to the controller. I don't think it should do the `append` but instead we need to do the `append` in the `drop_` method.